otter/test/convergence/test_service.py

```diff
@@ -1366,6 +1374,41 @@ def plan(*args, **kwargs):
             perform_sequence(self.get_seq() + sequence, self._invoke(plan)),
             ConvergenceIterationStatus.Stop())
 
+    def test_launch_stack_config(self):
+        """Using a launch_stack launch config works."""
+        lc = {'args': {'stack': {'stack_name': 'foo'}},
+              'type': 'launch_stack'}
+
+        self.manifest = {
+            'state': self.state,
+            'launchConfiguration': lc,
+        }
+
+        self.gsgi_result = (self.group, self.manifest)
+        self.gacd_runner = lambda i: {'stacks': []}
+
+        def plan(*args, **kwargs):
+            return []
+
+        executor = attr.assoc(launch_stack_executor,
+                              gather=intent_func('gacd'),
+                              plan=plan)
+
+        seq = [
+            parallel_sequence([]),  # No steps to log (log_steps)
+            (Log(msg='execute-convergence', fields=mock.ANY), noop),
+            (Log(msg='execute-convergence-results', fields=mock.ANY), noop),
+            (ModifyReference(self.waiting,
+                             match_func(pmap({self.group_id: 43}),
+                                        pmap())),
+             dispatch(reference_dispatcher)),
+        ]
+        eff = execute_convergence(
+            self.tenant_id, self.group_id, build_timeout=3600,
+            waiting=self.waiting, limited_retry_iterations=43,
+            get_executor=lambda _: executor)
+        perform_sequence(self.get_seq(with_cache=False) + seq, eff)
```

Comment:
Would be nice to ensure that return value is `ConvergenceIterationStatus.Stop()`.